errctx.go

```diff
@@ -20,21 +21,26 @@ func ErrWithKV(err error, kvs ...KV) error {
 	if existingKV != nil {
 		kv = Merge(existingKV.(KV), kv)
 	}
-	return errctx.Set(err, kvKey(0), kv)
+	return errctx.MarkSkip(errctx.Set(err, kvKey(0), kv), 1)
```

Comment:
Should this just be 
```suggestion
	return errctx.Mark(errctx.Set(err, kvKey(0), kv))
```

since [`Mark`](https://github.com/levenlabs/errctx/blob/master/errctx.go#L78) is effectively `MarkSkip(err, 1)`?

Reply:
No it needs to be `MarkSkip` here because we want to ignore the call to `ErrWithKV` in the stack. If we called `Mark` it would call `MarkSkip(,1)` which would skip the `Mark` call and source this line but we want to skip this line and source the caller instead.